### DISCUSSION TITLE
PowerAutomateとメタデータ感情分析APIの連携

### DISCUSSION THREAD

**Discussion by Kimi1014** [1ft-seabass/ma-hackathon-qa-and-knowledge-202210#16] [Q&A] [answered]
PowerAutomateから感情解析APIにテキストを投げて、返ってくるスコアを変数に受け取りたいのですが、実装方法がわからないまま時間が経っています😭
実装事例やサンプルあれば教えていただきたいです！

**Comment by 1ft-seabass** [comment]
@Kimi1014 わたしも明確な経験ないんですが、みてみますねー。

たぶん、

- PowerAutomate で POST 送信するはず
- その仕様に則って感情解析APIのやり方に当てはめる

と予感します！

良い感じの文献あったらおしらせしまーす。

**Comment by mochan-tk** [comment, marked as answer]
@Kimi1014 
横から失礼いたします。
いったん簡単にですが、外部のAPIを実行する時は下記のような形で使えます。
あとは上にも書いていただいてますが、APIの仕様に則う形になります。

![image](https://user-images.githubusercontent.com/9846618/200213768-eca291f6-cad6-47ff-9a11-858bdad219f7.png)

**Reply by 1ft-seabass** [reply]
ほぼほぼ同じことをお伝えしようと思ってたので、ありがとうございます！ @mochan-tk

**Reply by Kimi1014** [reply]
ありがとうございます！助かります！
実装・動作確認してみます〜！

**Reply by Kimi1014** [reply]
以下のような設定でやってみます！

元のアクセス
http://ap.mextractr.net/ma9/emotion_analyzer?apikey="APIキー値"&text=”テキスト”

Automate
・URIに「http://ap.mextractr.net/ma9/emotion_analyzer」
・ヘッダー欄の「キーの入力」に「apikey」、「text」
・「値の入力」に"APIキー値"、”テキスト”

慣れていない分野で、間違っていたらご指摘いただけたら助かります。。

**Reply by Kimi1014** [reply]
あ、textは本文ですかね。。
メタデータさんにも聞いてみます。。

**Reply by 1ft-seabass** [reply]
@Kimi1014 

方法：　GET
URI：　http://ap.mextractr.net/ma9/emotion_analyzer?apikey=自分のAPIKEY&text=何かテキスト

だけで、ちょっと試してもらえますでしょうか？ヘッダー、本文はなにつけなくていいかもです。

ヘッダー：（何もつけなくていい）
本文：（何もつけなくていいかも）

**Reply by 1ft-seabass** [reply]
http://ap.mextractr.net/ma9/emotion_analyzer?apikey=自分のAPIKEY&text=何かテキスト

は、ブラウザのアドレスバーからでも実験できます。自分のAPIKEYは自分のAPIKEYを入れてください。 @Kimi1014

**Reply by Kimi1014** [reply]
ありがとうございます！試してみます。

**Reply by 1ft-seabass** [reply]
XML で返る件は http://ap.mextractr.net/ma9/emotion_analyzer?apikey=APIKEY&text=何かテキスト&out=json と、末尾に &out=json 加えると良いですー。 @Kimi1014

**Reply by 1ft-seabass** [reply]
@Kimi1014 こちら、うまくすすみましたかー？

なにか不都合あればコメントください。

**Reply by Kimi1014** [reply]
ありがとうございます！おかげさまで、何とかうまくいっております！感謝です、、！

**Reply by 1ft-seabass** [reply]
おおおよかったです！
